test/easy-stackup_test.js

```diff
@@ -0,0 +1,145 @@
+    easyStackup(layers, function(error, stackup) {
+      expect(error).to.not.be.ok
+      expect(stackup.layers[0].options.id).to.equal('test-id')
+      done()
+    })
+  })
+  it('should not overwrite stackup id', function(done) {
+    var layers = [{gerber:emptyGerber, layerType:'tcu'}]
+    easyStackup(layers, {id: 'test-id'}, function(error, stackup) {
+      expect(error).to.not.be.ok
+      expect(stackup).to.be.ok
+      //TODO: test for actual id
+      done()
+    })
+  })
+  it('should callback with top, bottom and layer array', function(done) {
+    var layers = [{gerber:emptyGerber, filename: '', layerType:''}]
+    easyStackup(layers, function(error, stackup) {
+      expect(error).to.not.be.ok
+      expect(stackup).to.be.ok
+      expect(stackup).to.be.an('object')
+      expect(stackup).to.have.all.keys('top', 'bottom', 'layers')
+      expect(stackup.layers).to.be.an.instanceOf(Array)
+      expect(stackup.layers[0]).to.have.all.keys('layerType', 'gerber', 'options')
+      done()
+    })
+  })
+  it('respects plotAsOutline option', function(done) {
+    var layers = [{gerber:emptyGerber, layerType:'tcu', options:{plotAsOutline:true}}]
+    easyStackup(layers, function(error, stackup) {
+      expect(error).to.not.be.ok
+      expect(stackup.layers[0].options.plotAsOutline).to.be.true
+      done()
+    })
+  })
+  it('handles multiple layers', function(done) {
+    var layers = [{gerber:emptyGerber, layerType:'fcu'}, {gerber:emptyGerber, layerType:'tcu'}]
+    easyStackup(layers, function(error, stackup) {
+      expect(error).to.not.be.ok
+      expect(stackup).to.be.ok
+      expect(stackup.layers.length).to.equal(layers.length)
+      done()
+    })
+  })
+  it('can be passed back its own output', function(done) {
+    var layers = [
+      {gerber:emptyGerber, layerType:'fcu'},
+      {gerber:emptyGerber, layerType:'tcu'}
+    ]
+    easyStackup(layers, function(error, stackup) {
+      expect(error).to.not.be.ok
+      expect(stackup).to.be.ok
+      done()
+    })
+  })
+  //these determinism tests should really be property based using a quickcheck
+  //style framework instead of single unit tests
+  it('has deterministic top and bottom svgs if ids are given', function(done) {
+    var layers = [
+      {gerber:emptyGerber, layerType:'fcu', options:{id:'a'}},
+      {gerber:emptyGerber, layerType:'tcu', options:{id:'b'}}
+    ]
+    easyStackup(layers, {id:'c'}, function(error, stackup1) {
+      expect(error).to.not.be.ok
+      expect(stackup1).to.be.ok
+      easyStackup(layers, {id:'c'}, function(error, stackup2) {
+        expect(error).to.not.be.ok
+        expect(stackup2.top).to.deep.equal(stackup1.top)
+        expect(stackup2.bottom).to.deep.equal(stackup2.bottom)
+        done()
+      })
+    })
+  })
+  it('has deterministic top and bottom svgs if ids are given and passed back its own output', function(done) {
+    var layers = [
+      {gerber:emptyGerber, layerType:'fcu', options:{id:'a'}},
```

Comment:
Doesn't affect anything right now but might in the future, so will change it. I was thinking about enforcing correct layer types actually but gave up because behaviour so far has been not to care. 

https://github.com/tracespace/pcb-stackup/pull/12/commits/501366e7734fccd43cc66fa44bcf98a27e0cf8e8